### PR TITLE
[feat] Prepare animated text kit flutter web app for demo

### PR DESCRIPTION
**Pull request by aagarwal1012**
### Tickets
- Closes #222 

### Description
- Setup firebase hosting for `example` web app.
- Added GitHub actions workflow files for `deployment preview` and `production deployment` to firebase hosting.
- Live `web-app` is accessible at https://animated-text-kit.web.app.
- Sent invitations for the firebase project access to @awhitford and @SirusCodes.

### CODE REVIEW

File: .github/workflows/deploy-preview.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        working-directory: ./example
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
+        with:
+          java-version: "12.x"
+      - uses: subosito/flutter-action@v1
+        with:
+          channel: stable
```
Comment (awhitford) [inline .github/workflows/deploy-preview.yml:22]: I think the `stable` channel is the _default_, so you don't really need to be explicit.

:rocket: Advanced would be to offer a _Workflow Input Parameter_ that defaults to `stable`, but can be overridden to be another channel, or a version, and then use that.

File: .github/workflows/deploy-preview.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  preview:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./example
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
```
Comment (awhitford) [inline .github/workflows/deploy-preview.yml:17]: Setting up Java should only be important when you are building for Android -- not targeting Web.

And if it is necessary, why version 12?  (I'm thinking that an LTS version should be used.). My guess is that you copied/pasted this from the `flutter-action` `README`.

File: .github/workflows/deploy-preview.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+          java-version: "12.x"
+      - uses: subosito/flutter-action@v1
+        with:
+          channel: stable
+      - run: flutter pub get
+      - run: flutter build web --release
+      - uses: FirebaseExtended/action-hosting-deploy@v0
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_ANIMATED_TEXT_KIT }}"
```
Comment (awhitford) [inline .github/workflows/deploy-preview.yml:28]: 💭 `FIREBASE_SERVICE_ACCOUNT_ANIMATED_TEXT_KIT` is long.
Is qualifying a _project secret_ with the project name (`_ANIMATED_TEXT_KIT`) really necessary? 🤔

File: .github/workflows/deploy-prod.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+name: Deploy to Live Channel
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  preview:
```
Comment (awhitford) [inline .github/workflows/deploy-prod.yml:11]: Maybe this job name should be "live" instead of "preview."

File: .github/workflows/deploy-prod.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./example
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v1
```
Comment (awhitford) [inline .github/workflows/deploy-prod.yml:19]: Same issue as above; I don't think setting up Java is necessary.

File: example/web/manifest.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "example",
+    "short_name": "example",
+    "start_url": ".",
+    "display": "standalone",
+    "background_color": "#0175C2",
+    "theme_color": "#0175C2",
+    "description": "A new Flutter project.",
```
Comment (awhitford) [inline example/web/manifest.json:8]: Let's say _Animated Text Kit_ or something.

File: example/web/manifest.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{
+    "name": "example",
```
Comment (awhitford) [inline example/web/manifest.json:2]: Maybe we should say `Animated Text Kit Example`?

**Review by awhitford — CHANGES_REQUESTED** [review]
This works pretty well and I really like it! 🚀 
* One issue with the preview is that the default text color is black instead of white.  I think this causes Wavy to show nothing (black on black).  This is not necessarily an issue with this commit -- it is likely more of a web issue, but I'm pointing it out in case there is a tweak opportunity, like the `web/manifest.json` file.
* The _Next_ button on the **Colorize** page needs to be clicked twice.  I suspect that there is a stack trace to be investigated.
* Some of the commit activity looks related to Android, rather than Web.

File: example/web/manifest.json (hunk 7 above)
Comment (aagarwal1012) [reply example/web/manifest.json]: Fixed.

File: example/web/manifest.json (hunk 6 above)
Comment (aagarwal1012) [reply example/web/manifest.json]: Fixed.

File: .github/workflows/deploy-prod.yml (hunk 5 above)
Comment (aagarwal1012) [reply .github/workflows/deploy-prod.yml]: Fixed!

File: .github/workflows/deploy-prod.yml (hunk 4 above)
Comment (aagarwal1012) [reply .github/workflows/deploy-prod.yml]: Fixed!

File: .github/workflows/deploy-preview.yml (hunk 3 above)
Comment (aagarwal1012) [reply .github/workflows/deploy-preview.yml]: Actually, I also feel the same but this was the recommended way, but I have updated the secret name to `FIREBASE_SERVICE_ACCOUNT`.

File: .github/workflows/deploy-preview.yml (hunk 2 above)
Comment (aagarwal1012) [reply .github/workflows/deploy-preview.yml]: Fixed!

File: .github/workflows/deploy-preview.yml (hunk 1 above)
Comment (aagarwal1012) [reply .github/workflows/deploy-preview.yml]: Fixed!

File: example/web/index.html [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  <!-- iOS meta tags & icons -->
+  <meta name="apple-mobile-web-app-capable" content="yes">
+  <meta name="apple-mobile-web-app-status-bar-style" content="black">
+  <meta name="apple-mobile-web-app-title" content="example">
+  <link rel="apple-touch-icon" href="icons/Icon-192.png">
+
+  <!-- Favicon -->
+  <link rel="icon" type="image/png" href="favicon.png"/>
+
+  <title>example</title>
```
Comment (awhitford) [inline example/web/index.html:29]: Animated Text Kit Example?

File: .github/workflows/deploy-preview.yml [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./example
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: subosito/flutter-action@v1
+      - run: flutter pub get
+      - run: flutter build web --release
```
Comment (awhitford) [inline .github/workflows/deploy-preview.yml:19]: Minor suggestion:  Style-wise, I like the multi-line approach like:
```yaml
        run: |
          flutter pub get
          flutter build web --release
```

File: .github/workflows/deploy-prod.yml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./example
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: subosito/flutter-action@v1
+      - run: flutter pub get
+      - run: flutter build web --release
```
Comment (awhitford) [inline .github/workflows/deploy-prod.yml:21]: Similar style suggestion as above.

**Review by awhitford — CHANGES_REQUESTED** [review]
One change (`title`), and two minor suggestions.

**Comment by aagarwal1012** [comment]
> One issue with the preview is that the default text color is black instead of white. I think this causes Wavy to show nothing (black on black). This is not necessarily an issue with this commit -- it is likely more of a web issue, but I'm pointing it out in case there is a tweak opportunity, like the web/manifest.json file.  
- Yes, I have fixed this issue, changed the wavy text background-color to `purple`.
</br>

> The Next button on the Colorize page needs to be clicked twice. I suspect that there is a stack trace to be investigated.
- I am not able to debug this issue, it says that `Another exception was thrown: UnimplementedError`.
</br>

> Some of the commit activity looks related to Android, rather than Web.
- When I was setting up the example app for the web, `flutter create .` created some extra files in the `Android` folder which seems to be necessary.
</br>

> Should there be a link to the latest live page on the README?
- Done!
</br>

> Is it possible to embed the version (like in the footer) to be explicit?
- I think this will not be useful because we already have a version badge in the Readme. Generally, app versions should be displayed in the footer, but in the case of `example` app, we don't maintain any version number.

File: .github/workflows/deploy-prod.yml (hunk 10 above)
Comment (aagarwal1012) [reply .github/workflows/deploy-prod.yml]: Fixed!

File: .github/workflows/deploy-preview.yml (hunk 9 above)
Comment (aagarwal1012) [reply .github/workflows/deploy-preview.yml]: Fixed!

File: example/web/index.html (hunk 8 above)
Comment (aagarwal1012) [reply example/web/index.html]: Done👍

File: README.md [hunk 11]
```diff
@@ -2,7 +2,7 @@
 
 <h1 align="center">Animated Text Kit</h1>
 
-<p align="center">A flutter package which contains a collection of some cool and awesome text animations. Recommended package for text animations in Codemagic's Ebook, <a href="https://blog.codemagic.io/flutter-libraries-ebook-by-codemagic/ebook-flutter-libraries-we-love-by-codemagic.pdf">"Flutter libraries we love"</a>.</p><br>
+<p align="center">A flutter package which contains a collection of some cool and awesome text animations. Recommended package for text animations in Codemagic's Ebook, <a href="https://blog.codemagic.io/flutter-libraries-ebook-by-codemagic/ebook-flutter-libraries-we-love-by-codemagic.pdf">"Flutter libraries we love"</a>. Try-out our example app in <a href="https://animated-text-kit.web.app/">live action</a>. 🚀🚀</p><br>
```
Comment (awhitford) [inline README.md:6]: Minor:  I would say, "Try out our [live example app]."

**Review by awhitford — APPROVED** [review]
I have one minor suggestion for the `README` link, but otherwise this looks great.

File: README.md (hunk 11 above)
Comment (aagarwal1012) [reply README.md:6]: Fixed!

**Comment by aagarwal1012** [comment]
Thanks, @awhitford for the thorough review! Now I'm merging this PR.

**Comment by awhitford** [comment]
> * The _Next_ button on the **Colorize** page needs to be clicked twice.  I suspect that there is a stack trace to be investigated.

The stack trace looks like:
```
══╡ EXCEPTION CAUGHT BY SCHEDULER LIBRARY ╞═════════════════════════════════════════════════════════
The following UnimplementedError was thrown during a scheduler callback:
UnimplementedError

When the exception was thrown, this was the stack:
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 236:49  throw_
lib/_engine/engine/canvaskit/layer_scene_builder.dart 196:5                   pushShaderMask
packages/flutter/src/rendering/layer.dart 1779:26                             addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1966:5                              addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1131:5                              addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1131:5                              addToScene
packages/flutter/src/rendering/layer.dart 407:5                               [_addToSceneWithRetainedRendering]
packages/flutter/src/rendering/layer.dart 994:14                              addChildrenToScene
packages/flutter/src/rendering/layer.dart 1576:5                              addToScene
packages/flutter/src/rendering/layer.dart 732:5                               buildScene
packages/flutter/src/rendering/view.dart 217:30                               compositeFrame
packages/flutter/src/rendering/binding.dart 457:18                            drawFrame
packages/flutter/src/widgets/binding.dart 874:13                              drawFrame
packages/flutter/src/rendering/binding.dart 319:5                             [_handlePersistentFrameCallback]
packages/flutter/src/scheduler/binding.dart 1144:15                           [_invokeFrameCallback]
packages/flutter/src/scheduler/binding.dart 1082:9                            handleDrawFrame
packages/flutter/src/scheduler/binding.dart 998:5                             [_handleDrawFrame]
lib/_engine/engine/platform_dispatcher.dart 896:13                            invoke
lib/_engine/engine/platform_dispatcher.dart 145:5                             invokeOnDrawFrame
lib/_engine/engine.dart 259:45                                                <fn>
════════════════════════════════════════════════════════════════════════════════════════════════════
Another exception was thrown: UnimplementedError
Another exception was thrown: UnimplementedError
Another exception was thrown: UnimplementedError
```

I think this issue is related to https://github.com/flutter/flutter/issues/75495 -- but that appears to be fixed already, and _should_ be part of `stable`.

**Pull request merged**